### PR TITLE
chore(test): add http strategy tests

### PR DESCRIPTION
**Pull request by martinschaer**
## ☕️ Reasoning

Surrealdb.js supports socket and HTTP connections. This PR includes unit tests for both types of connections.

## 🧢 Checklist

- [ ] Wait for surrealdb.js to merge this PR that fixes query variables on HTTP: https://github.com/surrealdb/surrealdb.js/pull/171
- [ ] Documentation
- [x] Tests
- [ ] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

Fixes: INSERT_ISSUE_LINK_HERE

### CODE REVIEW

File: packages/adapter-surrealdb/tests/index.test.ts [hunk 1]
```diff
@@ -1,18 +1,19 @@
-import Surreal from "surrealdb.js"
+import Surreal, { ExperimentalSurrealHTTP } from "surrealdb.js"
 import { runBasicTests } from "@auth/adapter-test"
+import fetch from "node-fetch"
```
Comment (ThangHuuVu) [inline packages/adapter-surrealdb/tests/index.test.ts:3]: can we not use node-fetch here? tests were passing for me even without it.

Comment (martinschaer) [reply packages/adapter-surrealdb/tests/index.test.ts:3]: For this PR we need to wait for surrealdb.js to merge this other PR: https://github.com/surrealdb/surrealdb.js/pull/171

I tested it using my fork with the fix

Comment (balazsorban44) [reply packages/adapter-surrealdb/tests/index.test.ts:3]: `node-fetch` (or any other `fetch` polyfill) indeed is not necessary anymore, as all Node.js versions have `fetch` already available.

**Comment by martinschaer** [comment]
Can we close this PR in favor of this one? https://github.com/nextauthjs/next-auth/pull/8823

Sorry for the mess...

Also, looks like node 18.18.0 still requires node-fetch, so I kept it for now

**Pull request closed without merging**